packages/core-data/src/actions.js

```diff
@@ -525,8 +525,9 @@ export const saveEntityRecord =
 		if ( ! entityConfig ) {
 			return;
 		}
-		const entityIdKey = entityConfig.key || DEFAULT_ENTITY_KEY;
+		const entityIdKey = entityConfig.key ?? DEFAULT_ENTITY_KEY;
 		const recordId = record[ entityIdKey ];
+		const isNewRecord = !! entityIdKey && ! recordId;
```

Comment:
Why are we concerned about `entityIdKey` here?

Reply:
For the `root/site`, it will be intentionally false - https://github.com/WordPress/gutenberg/pull/72666#discussion_r2460576993. It's more like a safeguard.

Reply:
I don't expect this to break any consumer code.